src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -160,6 +160,13 @@ def _populate_core_stats_and_sim_state(self, area):
             core_stats_dict['grid_fee_constant'] = area.current_market.const_fee_rate \
                 if area.current_market is not None else 0.
 
+        if isinstance(area.strategy, HomeMeterStrategy):
+            core_stats_dict["home_meter_profile_kWh"] = (
+                area.strategy.state.get_energy_at_market_slot(self.current_market_time_slot))
+            if area.parent.current_market is not None:
+                for trade in area.strategy.trades[area.parent.current_market]:
+                    core_stats_dict["trades"].append(trade.serializable_dict())
```

Comment:
These 3 lines are repeated 6 times below, would be great if it can exit the if conditions and be set at the last, maybe inside a BaseStrategy condition to make sure it won't be called on invalid input

Reply:
I agree that it would be good to refactor this somehow. I'm not sure that the BaseStrategy should be changed though because it would mean allowing something like `strategy.trades[None]`, which I find counterintuitive.
I personally think that the entire `SimulationEndpointBuffer` class should be refactored and split into several classes (in particular for the `_populate_core_stats_and_sim_state` method). But if that's ok for you I would keep it out of the current PR since it might take some time.

Reply:
I agree with all of the above, just for the future IMO the BaseStrategy should not offer a full implementation, just the method with an Exception to notify us in case it is not implemented for a strategy, and each of these if blocks should belong to the respective strategy classes (in order to avoid the isinstance checks)